pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -281,11 +288,19 @@ func (ds *dockerService) getIP(sandbox *dockertypes.ContainerJSON) (string, erro
 		// reporting the IP.
 		return "", nil
 	}
-	if IP, err := ds.getIPFromPlugin(sandbox); err != nil {
-		glog.Warningf("%v", err)
-	} else if IP != "" {
-		return IP, nil
+
+	// Don't bother getting IP if networking isn't ready
```

Comment:
@yujuhong good point; though it's hard to use namespace/name because we can't easily get that from RemovePodSandbox() which is where we actually need to delete the item from the networkReady map.  We need to delete there because the testcases rightfully point out that after StopPodSandbox() a status request shouldn't return the Pod IP, so if we delete from the map in StopPodSandbox() then "if key does not exist, get the IP as usual" breaks that testcase.

And you're right, we don't get the sandbox ID until the container is created.  The real race is between ListPodSandbox() and RunPodSandbox(), where there is a "large" amount of time between creating the docker container and finishing the network setup, while in between those operations PLEG has access to the sandbox ID via ListPodSandbox().

So I suggest a hybrid approach.  We still use the pod sandbox ID as the key, but we add to the networkReady map with 'false' right after creation in RunPodSandbox(), then switch to 'true' when network has completed.  We flip back to 'false' in StopPodSandbox(), and delete from the map in RemovePodSandbox().

Then we use your suggestion for GetPodSandboxStatus() where if the key is not present, we assume network ready.  That should handle kubelet restart by allowing the status to proceed, but block PLEG for pods not-yet-setup.

Seem OK?